sfepy/discrete/common/region.py

```diff
@@ -676,19 +676,62 @@ def setup_mirror_region(self):
         """
         Find the corresponding mirror region, set up element mapping.
         """
+        from sfepy.discrete.fem.mesh import find_map
+
         regions = self.domain.regions
+        eopts = self.extra_options
+
+        if self.mirror_region is not None:
+            return
 
-        for reg in regions:
-            mirror_parent = regions.find(reg.parent)
-            if mirror_parent is None: continue
-            if ((reg is not self)
-                and nm.all(self.vertices == reg.vertices)):
-                mirror_region = reg
-                break
+        if (eopts is not None) and ('mirror_region' in eopts):
+            mreg = regions[eopts['mirror_region']]
+            if self.vertices.shape[0] != mreg.vertices.shape[0]:
+                raise ValueError('%s: incompatible mirror region! (%s)'
+                    % (self.name, mreg.name))
+            coors = self.domain.cmesh.coors
+            coors1 = coors[self.vertices, :]
+            coors2 = coors[mreg.vertices, :]
+            shift = ((nm.sum(coors2, axis=0) - nm.sum(coors1, axis=0))
+                / coors1.shape[0])
+            vmap1, vmap2 = find_map(coors1, coors2 - shift, join=False)
+            if vmap1.shape[0] != coors1.shape[0]:
+                print(coors1[vmap1])
+                print(coors2[vmap2])
+                raise ValueError('cannot match vertices!')
+
+            fconn = self.domain.cmesh.get_conn_as_graph(self.dim - 1, 0)
+            # region facet centers
+            fcc = nm.empty((self.facets.shape[0], self.dim), dtype=nm.float64)
```

Comment:
Maybe just use CMesh.get_centroids() which computes all facet centroids? I do not think the overhead would be that much. Anyway, get_centroids(region) is generally useful and should be available as a function.

Reply:
OK, I did not remember this function.